mysql/data_test.go

```diff
@@ -0,0 +1,290 @@
+			}})
+	conv.SetDataMode()
+	var rows []spannerData
+	conv.SetDataSink(
+		func(table string, cols []string, vals []interface{}) {
+			rows = append(rows, spannerData{table: table, cols: cols, vals: vals})
+		})
+	ProcessDataRow(conv, tableName, cols, conv.SrcSchema[tableName], tableName, cols, conv.SpSchema[tableName], []string{"4.2", "6", "prisoner zero"})
+	assert.Equal(t, []spannerData{spannerData{table: tableName, cols: cols, vals: []interface{}{float64(4.2), int64(6), "prisoner zero"}}}, rows)
+}
+
+func TestConvertData(t *testing.T) {
+	singleColTests := []struct {
+		name    string
+		ty      ddl.ScalarType
+		isArray bool
+		srcTy   string      // Source DB type (Used by e.g. timestamp conversions).
+		in      string      // Input value for conversion.
+		e       interface{} // Expected result.
+	}{
+		{"bool", ddl.Bool{}, false, "", "1", true},
+		{"bytes", ddl.Bytes{Len: ddl.MaxLength{}}, false, "", string([]byte{137, 80}), []byte{0x89, 0x50}}, // need some other approach to testblob type
+		{"date", ddl.Date{}, false, "", "2019-10-29", getDate("2019-10-29")},
+		{"float64", ddl.Float64{}, false, "", "42.6", float64(42.6)},
+		{"int64", ddl.Int64{}, false, "", "42", int64(42)},
+		{"string", ddl.String{Len: ddl.MaxLength{}}, false, "", "eh", "eh"},
+		{"datetime", ddl.Timestamp{}, false, "datetime", "2019-10-29 05:30:00", getTimeWithoutTimezone(t, "2019-10-29 05:30:00")},
+		{"timestamp", ddl.Timestamp{}, false, "timestamp", "2019-10-29 05:30:00", getTime(t, "2019-10-29T05:30:00+05:30")},
+		{"string array(set)", ddl.String{Len: ddl.MaxLength{}}, true, "", "1,Travel,3,Dance", []spanner.NullString{
+			spanner.NullString{StringVal: "1", Valid: true},
+			spanner.NullString{StringVal: "Travel", Valid: true},
+			spanner.NullString{StringVal: "3", Valid: true},
+			spanner.NullString{StringVal: "Dance", Valid: true}}},
+	}
+	tableName := "testtable"
+	for _, tc := range singleColTests {
+		col := "a"
+		conv := buildConv(
+			ddl.CreateTable{
+				Name:     tableName,
+				ColNames: []string{col},
+				ColDefs:  map[string]ddl.ColumnDef{col: ddl.ColumnDef{Name: col, T: tc.ty, IsArray: tc.isArray, NotNull: false}},
+				Pks:      []ddl.IndexKey{}},
+			schema.Table{Name: tableName, ColNames: []string{col}, ColDefs: map[string]schema.Column{col: schema.Column{Type: schema.Type{Name: tc.srcTy}}}})
+		conv.TimezoneOffset = "+05:30"
+		at, ac, av, err := ConvertData(conv, tableName, []string{col}, conv.SrcSchema[tableName], tableName, []string{col}, conv.SpSchema[tableName], []string{tc.in})
+		checkResults(t, at, ac, av, err, tableName, []string{col}, []interface{}{tc.e}, tc.name)
+	}
+
+	timestampTests := []struct {
+		name  string
+		srcTy string
+		in    string
+		e     interface{}
+	}{
+		{"timestampt", "timestamp", "2019-10-29 05:30:00", getTime(t, "2019-10-29T05:30:00+10:00")},
+		{"datetime", "datetime", "2019-10-29 05:30:00", getTimeWithoutTimezone(t, "2019-10-29 05:30:00")},
+	}
+	for _, tc := range timestampTests {
```

Comment:
Please consider to use `t.Run` to organize sub-tests: https://github.com/golang/go/wiki/TableDrivenTests#example-of-a-table-driven-test.

Reply:
Done. Added `t.Run()` in `mysqldump_test.go` and `data_test.go`